frontend/components/panels/NotifCheckBox.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of Search NEU and is licensed under AGPL3.
+ * See the license file in the root folder for details.
+ */
+
+import React from 'react';
+import { Checkbox, Icon } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import user from '../user';
+import Keys from '../../../common/Keys';
+
+// This file renders the checkboxes that control which sections a user signs up for
+// notifications.
+export default class NotifCheckBox extends React.Component {
+  // the only thing required for you to pass in is what section this NotifCheckBox
+  // was rendered in. We can get all the other data from that.
+  static propTypes = {
+    section: PropTypes.object.isRequired,
+  }
+
+  // creates a new Notification Check Box that has the section's
+  // hash in section, and sets checked to be the initial value of if the section
+  // is currently in the user or not
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // has a user already signed up for notifications on a section?
+      checked: user.hasSectionAlready(Keys.getSectionHash(this.props.section)),
+
+      // the section for a class that this NotifCheckBox is for
+      section: this.props.section,
```

Comment:
Don't copy section to the state - throughout this component you want to just use this.props.section. 

Give these docs a read-down, they've got a lot of info about how react works :) 

https://reactjs.org/docs/components-and-props.html